src/numeric_filter.c

```diff
@@ -74,11 +74,18 @@ NumericFilter *NumericFilter_LegacyParse(ArgsCursor *ac, QueryError *status) {
 
   // Parse the min range
   const char *s = AC_GetStringNC(ac, NULL);
+  if (!s || !*s) {
```

Comment:
`!s` is always false here, as we previously checked that `AC_NumRemaining(ac) >= 3`, so we know we have strings here.
Also, note that `parseDoubleRange` cannot handle `NULL`s so you can be sure you're not exposing the code to any new risks here